pkg/roll/run_update.go

```diff
@@ -251,17 +250,23 @@ func (u update) lockRCs(done <-chan struct{}) error {
 // individual releases are successful or until the session is reset.
 func (u update) unlockRCs(done <-chan struct{}) {
 	wg := sync.WaitGroup{}
-	for _, path := range []string{u.lockPath(u.NewRC), u.lockPath(u.OldRC)} {
+	for _, rcID := range []rcf.ID{u.NewRC, u.OldRC} {
 		wg.Add(1)
-		go func(path string) {
+		go func(rcID rcf.ID) {
 			RetryOrQuit(
-				func() error { return u.lock.Unlock(path) },
+				func() error {
+					path, err := rcstore.RCUpdateLockPath(rcID)
```

Comment:
You're going to have to make the path outside the `RetryOrQuit()` loop, which keeps running the function until it returns a `nil` error.  As-is, an invalid path results in an infinite loop.